app/internal/date/date.go

```diff
@@ -0,0 +1,32 @@
+package date
```

Comment:
I want to improve this in a later PR. I'm thinking it should have Un/MarshalText methods so we can use it in structs where we want a date, and not a time. Also promote `After`/`Before` from `T` so we can hide the underlying `time.Time` value.